extlib/millhone/src/api/types.rs

```diff
@@ -0,0 +1,84 @@
+//! Concrete types for API communications, independent of API version.
+
+mod error;
+pub use error::*;
+
+use std::collections::HashMap;
+
+use derive_more::{AsRef, Display, From, Into};
+use itertools::intersperse;
+use serde::{Deserialize, Serialize};
+
+/// A dependency specified by the Millhone service.
```

Comment:
If it's not too hard, would `ServiceDependency` make this more obvious to consumers?

Reply:
Yeah, I waffled on naming it that. Ultimately I think `Service` doesn't _really_ clear up the confusion (it could still be interpreted as "dependency of the service being analyzed by FOSSA").

Ultimately, this type is in the `api` module, so its full path is `api::Dependency`, which _should_ make it clear: when considering Rust symbols, you're supposed to consider the full path, although some repetition is okay.

The comment is more just to fully clarify if a future reader still has a question.

Reply:
🤔 maybe types imported through `api::prelude` should all be prefixed with `Api` to ensure that distinction is carried through. I'll make that change.

Reply:
I think that works. One thing I thought of later too is that someone who really wants to know is likely using tooling that can bring them to your comment. So probably nbd.